File: patterns/c++/pattern22.cpp
#include <iostream>
using namespace std;

int main()
{
    int n;
    cin >> n;
    for (int i = -n + 1; i < n; i++)
    {
        for (int j = -n + 1; j < n; j++)
        {
            cout << (abs(i) > abs(j) ? abs(i) + 1 : abs(j) + 1);
        }
        cout << endl;
    }
    return 0;
}

File: patterns/Javascript/pattern22.js
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question('Enter a number:', (num) => {

    const n = parseInt(num);

    for(let i=-n+1;i<n;i++){
        console.log([...Array(2*n-1)].map((_,j) => Math.max(Math.abs(i),Math.abs(j-n+1))+1).join(""));
    }

    rl.close();
});